models/include/cloe/component/steering_actuator.hpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 Robert Bosch GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * \file cloe/component/steering_actuator.hpp
+ *
+ */
+
+#pragma once
+#ifndef CLOE_COMPONENT_STEERING_ACTUATOR_HPP_
+#define CLOE_COMPONENT_STEERING_ACTUATOR_HPP_
+
+#include <cloe/component/actuator.hpp>  // for Actuator
+
+namespace cloe {
+
+struct SteeringRequest {
```

Comment:
The `Actuator` interface lets you set all of the `SteeringRequest` or none of it.
How do you intend to communicate:
- Change the front wheel angle by 1 further degree?

Also, this interface opens the possibility of steering each wheel independantly.
I realize that there are vehicles that support this, but most do not.

Is this really the interface that a "steering request" should have?
I would at minimum name this `SteeringRequest4` or something else to indicate that it is 4-way steering.

Feel free to convince me otherwise.

Reply:
> Change the front wheel angle by 1 further degree?

- one could call get() method of the actuator in order to obtain the current values of "target_", then add +1 to the front wheel angle value and set it again. alternative would be to have a method that returns a reference of target_ or its address.

For the 4-way steering question i will try to get feedback and come back to you.

Reply:
Up until now, the semantics of actuator components have been:
- `target` is of type `optional<type>`
- If `target` is empty, no controller has overridden or set the value -- simulator driver model is in control
- If `target` is not empty, some controller has overridden or set the value
- With `get()`, a controller can check the last set value. If multiple systems set the brake actuator
- Every cycle, after the simulator plugin has applied the values, they are reset to empty.

Therefore, calling the `get()` method of the actuator in order to obtain the current value of the front wheel angle value is a deviation from how it's been handled right now.
So far, we've always needed a separate sensor.

Perhaps the current design is sub-optimal. We could also make the actuator into a sensor-actuator and include methods for `current_wheel_angle_vl` etc to make it easier to modify.
Then add some method for applying a delta wheel angle that does the same as you suggested.

What happens when a higher wheel angle is requested than the vehicle can actually support?
- Log a warning / error
- Ignore the actuation
- Set to maximum supported wheel angle
- Abort simulation

And what should happen when multiple wheel angles are supplied that contradict each other?
- front left = 2.3, front right = None
- front left = 2.3, front right = 3.1
- front left = 2.3, front right = 2.3

Reply:
i changed the interface to steering_angle and a utility function (as a free function) which then can calculate (under some assumption, i.e. low speed) the angle of front left and front right. i am currently not aware of any use case where the controller wants to control a single wheel independently, so i think that solution works good without the need for plausibilization inside the steering actuator.